app/models/user.php

```diff
@@ -78,4 +154,128 @@ public function isLoginValid()
     {
         return $this->is_login_valid;
     }
+
+    /**
+     * Gets the user's rank and converts it to it's string value
+     */
+    public function getRank()
+    {
+        return self::$rankEquivalent[$this->rank];
+    }
+
+    /**
+     * Checks if the user's rank needs to be increased or decreased
+     */
+    public function updateRank($count)
+    {
+        $db = DB::conn();
+
+        $numRequired = $this->rank * self::RANKUP_MULTIPLIER;
+        $previousRankReq = ($this->rank-1) * self::RANKUP_MULTIPLIER;
+
+        if ($count >= $numRequired && $this->rank < self::MAX_RANK) {
+            $newRank = ++$this->rank;
+            $db->update('user', array('rank' => $newRank), array('id' => $this->id));
+        }
+
+        if ($count <= $previousRankReq && $this->rank > self::MIN_RANK) {
+            $newRank = --$this->rank;
+            $db->update('user', array('rank' => $newRank), array('id' => $this->id));            
+        }
+    }
+
+    /**
+     * Increments the user's comment count
+     */
+    public function addCommentCount()
+    {
+        $db = DB::conn();
+        $currentCount = $db->value('SELECT comment_count FROM user WHERE id = ?', array($this->id));
+        $newCount = ++$currentCount;
+        $db->update('user', array('comment_count' => $newCount), array('id' => $this->id));
+        $this->updateRank($newCount);
+    }
+
+    /**
+     * Decrements the user's comment count
+     */
+    public function subtractCommentCount()
+    {
+        $db = DB::conn();
+        $currentCount = $db->value('SELECT comment_count FROM user WHERE id = ?', array($this->id));
+        $newCount = --$currentCount;
+        $db->update('user', array('comment_count' => $newCount), array('id' => $this->id));
+        $this->updateRank($newCount);
+    }
+
+    /**
+     * Fetches the top users with the highest comment counts
+     */
+    public static function getTopTen()
+    {
+        $db = DB::conn();
+        $limit = TOP_LIMIT;
+        $query = "SELECT DISTINCT comment_count FROM user ORDER BY comment_count DESC LIMIT {$limit}";
```

Comment:
TOP_LIMIT is one of the constants declared in core.php po? Maybe better po to put it in this class instead if it won't be used by any other classes? That way we can fix this as well po :bow: